Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -296,6 +296,47 @@ open class PrependedBookmarkFolder: BookmarkFolder {
     }
 }
 
+open class ConcatenatedBookmarkFolder: BookmarkFolder {
+    fileprivate let main: BookmarkFolder
+    fileprivate let prepend: BookmarkFolder
+
+    init(main: BookmarkFolder, prepend: BookmarkFolder) {
+        self.main = main
+        self.prepend = prepend
+        super.init(guid: main.guid, title: main.guid)
+    }
+
+    override open var count: Int {
+        return main.count + prepend.count
+    }
+
+    override open subscript(index: Int) -> BookmarkNode? {
+        if index < prepend.count {
+            return prepend[index]
+        } else {
+            let adjustedIndex = index - prepend.count
+            return main[adjustedIndex]
+        }
+    }
+
+    override open func itemIsEditableAtIndex(_ index: Int) -> Bool {
+        if index < prepend.count {
+            return prepend.itemIsEditableAtIndex(index)
+        } else {
+            let adjustedIndex = index - prepend.count
+            return prepend.itemIsEditableAtIndex(adjustedIndex)
+        }
+    }
+
+    override open func removeItemWithGUID(_ guid: GUID) -> BookmarkFolder? {
+        guard let removedFolder = main.removeItemWithGUID(guid) else {
```

Comment:
Change the base removeItemWithGUID to not return nil? Seems like that will touch more code than I would like. How about just:
```
    override open func removeItemWithGUID(_ guid: GUID) -> BookmarkFolder? {
        let newMain = main.removeItemWithGUID(guid) ?? main
        let newAppend = append.removeItemWithGUID(guid) ?? append
        return ConcatenatedBookmarkFolder(main: newMain, append: newAppend)
    }
```